mapie/mondrian.py

```diff
@@ -0,0 +1,438 @@
+            if self.mapie_estimator.conformity_score is not None:
+                if not isinstance(self.mapie_estimator.conformity_score,
+                   self.allowed_regression_ncs):
+                    raise ValueError(
+                        "The conformity score for the MapieRegressor must " +
+                        f"be one of {self.allowed_regression_ncs}"
+                    )
+
+    def _check_fit_parameters(
+            self, X: ArrayLike, y: ArrayLike, groups: ArrayLike
+    ) -> Tuple[NDArray, NDArray, NDArray]:
+        """
+        Perform checks on the input data, groups and the estimator
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            The input data
+        y : ArrayLike of shape (n_samples,) or (n_samples, n_outputs)
+            The target values
+        groups : ArrayLike of shape (n_samples,)
+            The groups of individuals
+
+        Returns
+        -------
+        X : NDArray of shape (n_samples, n_features)
+            The input data
+        y : NDArray of shape (n_samples,) or (n_samples, n_outputs)
+            The target values
+        groups : NDArray of shape (n_samples,)
+        """
+        self._check_estimator()
+        self._check_mapie_classifier()
+        self._check_confomity_score()
+        X, y = indexable(X, y)
+        if isinstance(self.mapie_estimator, MapieMultiLabelClassifier):
+            y = _check_y(y, multi_output=True)
+        else:
+            y = _check_y(y)
+        X = cast(NDArray, X)
+        y = cast(NDArray, y)
+        groups = cast(NDArray, np.array(groups))
+        self._check_groups_fit(X, groups)
+
+        return X, y, groups
+
+    def _check_is_topk_calibrator(self):
+        """
+        Check that the predict_proba method can only be used with a
+        MapieCalibrator estimator
+        """
+        if not isinstance(self.mapie_estimator, MapieCalibrator):
+            raise ValueError(
+                "The predict_proba method can only be used with a " +
+                "MapieCalibrator estimator"
+            )
+
+    def _check_not_topk_calibrator(self):
+        """
+        Check that the predict method can only be used with a MapieCalibrator
+        estimator
+        """
+        if isinstance(self.mapie_estimator, MapieCalibrator):
+            raise ValueError(
+                "The predict method can only be used with a MapieClassifier," +
+                "MapieRegressor or MapieMultiLabelClassifier estimator"
+            )
+
+    def fit(self, X: ArrayLike, y: ArrayLike, groups: ArrayLike, **kwargs):
+        """
+        Fit the Mondrian method
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            The input data
+        y : ArrayLike of shape (n_samples,) or (n_samples, n_outputs)
+            The target values
+        groups : ArrayLike of shape (n_samples,)
```

Comment:
Force type List[int] to avoid checking this? Or mention it here.
Mention that each group must have at least 2 individuals.

Reply:
Mentioned in the docstring, as ArrayLikes are not necessarily Lists